tests/collections/reshape/testing_reshape.c

```diff
@@ -252,8 +242,6 @@ int main(int argc, char *argv[])
 
       ctp->arenas_datatypes[PARSEC_remote_no_re_reshape_DEFAULT_ADT_IDX]    = adt_default;
       ctp->arenas_datatypes[PARSEC_remote_no_re_reshape_LOWER_TILE_ADT_IDX] = adt_lower;
-      PARSEC_OBJ_RETAIN(adt_default.arena);
```

Comment:
Does this means we are not tracking the refcount on the arena anymore ? How does it get deallocated ?

Reply:
We still track the arena counts, but the adt is not destructed implicitly by the generated code, thus there is no need for the extra retain (which was to prevent said deallocation by the generated code)

Reply:
During my own review of these changes I found the following issue that contradict some aspect of my previous message: 
1. What I said above *is correct* for all codes that use parsec_matrix_adt_define_xyz, and codes that are using the DTD attach type, they have all the correct hooking points ready to call the symetric cleanup (and have to, because they have to free the actual MPI datatype they created). 
2. This is *not the case* for the `examples/Ex0?.jdf`, they use `parsec_arena_datatype_set_type` and use a predefined datatype to populate the static PTG adt arrays. Without the auto OBJ_DESTRUCT in the generated code, these will need to call an explicit cleanup function to be leak free. These should be *straightforward to adapt.*
* This is also incorrect for a number of PTG testers (e.g., touch.jdf, a2a.jdf, BT_reduction_wrapper, merge_sort_wrapper, rtt_wrapper, collections/reduce.c, all the ptgpp testers, 

We had a discussion about this on Friday during the team meeting
Q: because we don't auto cleanup the examples are now incorrect (they leak the arenas), should I (Aurelien) revert the change that removed the automatic OBJ_DESTRUCT on the static ADTs or do we prefer adapting the offending testers.
A: (Thomas): I like that the user has to make symmetrical calls, so users will need to call `parsec_arena_datatype_unset_type` (new function wrapping around OBJ_DESTRUCT, essentially)
A: (Aurelien): a subset of offenders in group 3. are missing the corresponding `_finalize` call to match their `xyz_initialize` and directly call `taskpool_free`, these require a larger change to adapt. 

Documenting the discussion so that @bosilca can react.